tests/integration/test_backup_restore_on_cluster/test_disallow_concurrency.py

```diff
@@ -188,39 +203,28 @@ def test_concurrent_backups_on_same_node():
 
     backup_name = new_backup_name()
 
-    id = (
+    # Backup #1.
+    id, status = (
         nodes[0]
         .query(f"BACKUP TABLE tbl ON CLUSTER 'cluster' TO {backup_name} ASYNC")
-        .split("\t")[0]
-    )
-
-    status = (
-        nodes[0]
-        .query(f"SELECT status FROM system.backups WHERE id == '{id}'")
         .rstrip("\n")
+        .split("\t")
     )
+
     assert status in ["CREATING_BACKUP", "BACKUP_CREATED"]
 
-    result, error = nodes[0].query_and_get_answer_with_error(
+    # Backup #2.
+    _, error = nodes[0].query_and_get_answer_with_error(
         f"BACKUP TABLE tbl ON CLUSTER 'cluster' TO {backup_name}"
     )
 
-    expected_errors = [
-        "Concurrent backups not supported",
-        f"Backup {backup_name} already exists",
-    ]
-    if not error:
-        wait_for_fail_backup(nodes[0], id, backup_name)
-
```

Comment:
The following two assertions were wrong - the first one was wrong if `error` is Null, and the second one was wrong if `error` is not Null. I fixed both assertions in this test and other tests, and also added comments and helper functions to make these tests easier to read.